src/pattern_matching/pssm/mod.rs

```diff
@@ -0,0 +1,209 @@
+// Copyright 2018 Kieran Hervold
+// Licensed under the MIT license (http://opensource.org/licenses/MIT)
+// This file may not be copied, modified, or distributed
+// except according to those terms.
+
+//! Implementation of position-specific scoring matrix (PSSM), aka position
+//! weight matrix (PWM), for both DNA and amino-acid sequences
+//!
+//! Matrices can either be built from example sequences, as shown below,
+//! or directly from a float matrix (type Array2<f32>).
+//!
+//! # Examples
+//!
+//! use bio::pattern_matching::pssm::DNAMotif;
+//! let pssm = DNAMotif::from(vec![
+//!            b"AAAA".to_vec(),
+//!            b"AATA".to_vec(),
+//!            b"AAGA".to_vec(),
+//!            b"AAAA".to_vec(),
+//!        ]);
+//! let start_pos = pssm.score(b"CCCCCAATA").unwrap().loc;
+//! println!("motif found at position {}", start_pos);
+
+use ndarray::prelude::Array2;
+use std::f32::NEG_INFINITY;
+use utils::IntoTextIterator;
+
+mod dnamotif;
+mod protmotif;
+
+pub use self::dnamotif::DNAMotif;
+pub use self::protmotif::ProtMotif;
+
+pub const EPSILON: f32 = 1e-5;
+pub const INVALID_MONO: u8 = 255;
+
+/// Errors
+#[derive(Debug, Clone)]
+pub enum PSSMError {
```

Comment:
Please use quick_error macro for this, see https://github.com/rust-bio/rust-bio/blob/66c7de44aae1c28780f1245e959a8127dd1f63ae/src/utils/interval.rs#L55